xcp_d/interfaces/utils.py

```diff
@@ -289,13 +289,25 @@ def _run_interface(self, runtime):
             qc_values_dict[entity.split('-')[0]] = entity.split('-')[1]
 
         # Calculate QC measures
-        mean_fd = np.mean(preproc_fd)
-        mean_fd_post_censoring = np.mean(postproc_fd)
+        mean_fd = np.nanmean(preproc_fd)
+        mean_fd_post_censoring = np.nanmean(postproc_fd)
         mean_relative_rms = np.nanmean(rmsd_censored)  # first value can be NaN if no dummy scans
-        mean_dvars_before_processing = np.mean(dvars_before_processing)
-        mean_dvars_after_processing = np.mean(dvars_after_processing)
-        fd_dvars_correlation_initial = np.corrcoef(preproc_fd, dvars_before_processing)[0, 1]
-        fd_dvars_correlation_final = np.corrcoef(postproc_fd, dvars_after_processing)[0, 1]
+        mean_dvars_before_processing = np.nanmean(dvars_before_processing)
+        mean_dvars_after_processing = np.nanmean(dvars_after_processing)
+
+        tmp_fd = preproc_fd.copy()
+        tmp_fd[np.isnan(tmp_fd)] = 0
+        tmp_dvars = dvars_before_processing.copy()
+        tmp_dvars[np.isnan(tmp_dvars)] = 0
+        fd_dvars_correlation_initial = np.corrcoef(tmp_fd, tmp_dvars)[0, 1]
+        del tmp_fd, tmp_dvars
+
+        tmp_fd = postproc_fd.copy()
+        tmp_fd[np.isnan(tmp_fd)] = 0
+        tmp_dvars = dvars_after_processing.copy()
+        tmp_dvars[np.isnan(tmp_dvars)] = 0
+        fd_dvars_correlation_final = np.corrcoef(tmp_fd, tmp_dvars)[0, 1]
```

Comment:
Annoying thought - do we need to tell the users how many measurements of dvars were included in the means? Or is it obvious from the other columns in the confounds?